content/influxdb/cloud-serverless/write-data/best-practices/optimize-writes.md

```diff
@@ -122,3 +143,248 @@ If a host's clocks isn't synchronized with NTP, timestamps may be inaccurate.
+
+1. In your terminal, enter the following command to create the sample data file. The command sets a `grace` variable that you'll use when configuring Telegraf in the next step.
+
+   ```bash
+   cat <<EOF > ./home.lp
+   home,room=Kitchen temp=23.1 1641063600
+   home,room=Kitchen hum=36.6 1641063600
+   home,room=Kitchen co=22i 1641063600
+   home,room=Living\ Room temp=22.7 1641063600
+   home,room=Living\ Room hum=36.4 1641063600
+   home,room=Living\ Room co=17i 1641063600
+   EOF
+   grace_seconds=$(($(date +%s)-1641063000))
+   grace="${grace_seconds}s"
+   ```
+
+2. Enter the following command to configure Telegraf to parse the file, merge the points, and write the data to a bucket. To merge series, you must specify the following in your Telegraf configuration:
+
+   - the timestamp precision for your input data (for example, `influx_timestamp_precision` for line protocol)
+   - Optional: `aggregators.merge.grace` to extend the window and allow more points to be merged. For demonstration purposes, the following example sets `grace` to a large duration to include the sample data timestamps.
+
+   <!--pytest-codeblocks:cont-->
+
+   ```bash
+   cat <<EOF > ./telegraf.conf
+   # Parse metrics from a file
+   [[inputs.file]]
+     ## A list of files to parse during each interval.
+     files = ["home.lp"]
+     ## The precision of timestamps in your data.
+     influx_timestamp_precision = "1s"
+     tagexclude = ["host"]
+   # Merge separate metrics that share a series key
+   [[aggregators.merge]]
+     grace = "$grace"
+     ## If true, drops the original metric.
+     drop_original = true
+   # Writes metrics as line protocol to the InfluxDB v2 API
+   [[outputs.influxdb_v2]]
+     ## InfluxDB credentials and the bucket to write data to.
+     urls = ["https://{{< influxdb/host >}}"]
+     token = "API_TOKEN"
+     organization = ""
+     bucket = "BUCKET_NAME"
+   EOF
+   ```
+
+3. Enter the following command to run Telegraf for one interval and then exit:
+
+   <!--pytest-codeblocks:cont-->
+
+   <!--before-test
+   ```bash
+   influx bucket create --name jason-test-create-bucket 2>/dev/null ||:
+   ```
+   -->
+
+   <!--pytest-codeblocks:cont-->
+
+   ```bash
+   # Run once and exit.
+   telegraf --once --config telegraf.conf
+   ```
+
+   Telegraf writes the following lines:
+
+   <!--pytest-codeblocks:cont-->
+
+   <!--hidden-test
+   ```bash
+   telegraf --test --config telegraf.conf
+   ```
+   -->
+
+   <!--pytest-codeblocks:expected-output-->
+
+   ```
+   > home,room=Kitchen co=22i,hum=36.6,temp=23.1 1641063600000000000
+   > home,room=Living\ Room co=17i,hum=36.4,temp=22.7 1641063600000000000
+   ```
+
+<!-- ### Avoid sending duplicate data
+  -- Can't get processors.dedup to work as expected.
+  -- The content for this section is in branch https://github.com/influxdata/docs-v2/compare/issue5155-preprocess-with-dedup
```

Comment:
I responded to your question in slack, but it was due to the timestamps you have on the data. The dedup processor takes into account the timestamp given the dedeup interval.